src/main/java/net/minecraftforge/fluids/BlockFluidBase.java

```diff
@@ -453,6 +453,13 @@ public boolean shouldSideBeRendered(@Nonnull IBlockState state, @Nonnull IBlockA
         return super.shouldSideBeRendered(state, world, pos, side);
     }
 
+    public boolean isCoveredWithFluid(IBlockAccess world, BlockPos pos)
+    {
+        IBlockState here = world.getBlockState(pos);
+        IBlockState up = world.getBlockState(pos.down(densityDir));
+        return here.getBlock() == this && (up.getMaterial().isLiquid() || up.getBlock() instanceof IFluidBlock);
+    }
```

Comment:
i just realized I responded to this in the wrong thread. I'll copy it here to make it easier on other reviewers. Sorry for the mix up.

> ![neighbour](https://cloud.githubusercontent.com/assets/10445063/23509330/a3c4f29e-ff55-11e6-874b-441bbca22991.png)
>
>The reason I'm testing for neighbours is becuase of situations like this. But you are right, as you can see in the screenshot, vanilla does not check for it. I would argue checking for it is actually more correct behaviour. But I also understand if it's more important to retain compatibility to vanilla mechanics. Should I get rid of the check?